Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.cpp

```diff
@@ -117,10 +117,20 @@ void DNSResolveQueueCFNet::performDNSLookup(const String& hostname, Ref<Completi
     nw_parameters_set_context(parameters.get(), context.get());
     RetainPtr resolver = adoptCF(nw_resolver_create_with_endpoint(hostEndpoint.get(), parameters.get()));
 
-    nw_resolver_set_update_handler(resolver.get(), dispatch_get_main_queue(), makeBlockPtr([resolver, completionHandler = WTFMove(completionHandler)] (nw_resolver_status_t status, nw_array_t resolvedEndpoints) mutable {
+    auto timer = makeUnique<Timer>([resolver, completionHandler]() {
+        WTF::ensureOnMainThread([resolver, completionHandler]() {
```

Comment:
Timer AFAIK needs to be created on main thread and is executing its callback on main thread.
This call to ensureOnMainThread is probably not needed.
Otherwise, we should use `resolver = WTFMove(resolver), completionHandler = WTFMove(completionHandler)` to save some refcounting (and making the outer lambda mutable.

Reply:
done.